unified-codes/libs/ui/src/lib/components/entity/organisms/EntityBrowser.stories.tsx

```diff
@@ -1,32 +1,340 @@
+    return filteredEntities.slice(start, end).sort((a: IEntity, b: IEntity) => {
+      if (a[orderBy] > b[orderBy]) return orderDesc ? -1 : 1;
+      if (b[orderBy] > a[orderBy]) return orderDesc ? 1 : -1;
+      return 0;
+    })
+   }, [entities, orderBy, orderDesc, rowsPerPage, page]);
+
+  const onSort = React.useCallback((_: string) => setOrderDesc(!orderDesc), [orderDesc, setOrderDesc]);
+
+  const onChangePage = React.useCallback((_: React.MouseEvent<HTMLButtonElement> | null, page: number) => 
+    setPage(page), [setPage]);
+
+  const onChangeRowsPerPage = React.useCallback((event: React.ChangeEvent<HTMLTextAreaElement | HTMLInputElement>) => 
+    setRowsPerPage(+event.target.value), [setRowsPerPage]);
+
+  const entityBrowserTableHeader = React.useMemo(() => (
+    <EntityTableHeader
+      classes={classes?.header}
+      columns={columns}
+      onSort={onSort}
+      orderDesc={orderDesc}
+      orderBy={orderBy}
+    />
+  ), [classes, columns, onSort, orderDesc, orderBy]);
+
+  const entityTableRows = React.useMemo(() => (
+    <React.Fragment>
+      {
+        data.map((entity: IEntity, index: number) => (
+          <EntityTableRow
+            classes={{
+              root: index % 2 ? classes?.table?.rowPrimary : classes?.table?.rowSecondary,
+              cell: classes?.table?.cell
+            }}
+            columns={columns}
+            entity={entity}
+            key={entity.code}
+          ></EntityTableRow>
+        ))
+      }
+    </React.Fragment>
+  ), [classes, columns, data]);
+    
+  const entityBrowserTable = React.useMemo(() => (
+    <EntityTable 
+      classes={classes?.table}
+      header={entityBrowserTableHeader}
+      rows={entityTableRows}
+    />
+  ), [classes, entityBrowserTableHeader, entityTableRows]);
+
+  const entityBrowserTablePagination = React.useMemo(() => (
+    <TablePagination
+      classes={{ root: classes?.tablePagination?.root }}
+      rowsPerPageOptions={rowsPerPageOptions}
+      component="div"
+      count={count}
+      rowsPerPage={rowsPerPage}
+      page={page}
+      onChangePage={onChangePage}
+      onChangeRowsPerPage={onChangeRowsPerPage}
+    />
+  ), [classes, rowsPerPageOptions, count, rowsPerPage, page, onChangePage, onChangeRowsPerPage]);
+
+  // Search bar.
+
+  const { 
+    input,
+    label,
+    setInput,
+  } = useSearchBarState();
+
+  const onChange = React.useCallback((input: string) => setInput(input), [setInput]);
+  const onSearch = React.useCallback((term: string) => setFilterBy(term), [setFilterBy]);
+  const onClear = React.useCallback(() => { setInput(''); setFilterBy(''); }, [setInput, setFilterBy]);
+
+  const entityBrowserSearchBar = React.useMemo(() => (
+    <SearchBar 
+      classes={classes.searchBar}
+      input={input}
+      label={label} 
+      onChange={onChange}
+      onClear={onClear}
+      onSearch={onSearch}
+    />
+  ), [classes, input, label, onChange, onClear, onSearch]);
+
+  // Type filter.
```

Comment:
Idea I was going for is that library components would be good as purely presentational, i.e. just "dumb" compositions of components which let the parent maintain all their state and just do what the parent says and tell the parent if an event occurs. 

I think `EntityTypeFilter` is uncontrolled (i.e. maintains internal state that parent can't see). Ideally, I'm thinking we should avoid any internal state management or hardcoded styles/child props (if we need to specify certain styles/ child props etc. can expose child directly as component prop so parent can take care of it).

So, what I'm thinking here to move logic up and have parent maintain toggle state (e.g. as part of a redux slice responsible for filter parameters, i.e. search term, ordering, types, etc.). Parent can then pass buttons directly as render/component props to ToggleButtonGroup.

@katherine-sussol Make sense? Agree? Disagree?